### PR TITLE
Add `push0`, bump instruction set to shanghai

### PR DESCRIPTION
**Pull request by lightclient**
Better late than never? This PR finally introduces `push0` and updates the default instruction set in all tools to Shanghai.

### CODE REVIEW

File: etk-analyze/Cargo.toml [hunk 1]
```diff
@@ -20,7 +20,7 @@ clap = { optional = true, version = "3.1", features = ["derive"] }
 etk-cli = { optional = true, path = "../etk-cli", version = "0.3.0-dev" }
 etk-asm = { optional = true, path = "../etk-asm", version = "0.3.0-dev" }
 etk-dasm = { path = "../etk-dasm", version = "0.3.0-dev" }
-z3 = "0.11.2"
+z3 = { version = "0.11.2", features = ["static-link-z3"] }
```
Comment (lightclient) [inline etk-analyze/Cargo.toml:23]: forgive me lord for i have sinned (err unable to dynamically link z3)

File: etk-analyze/.z3-trace
Comment (SamWilsn) [inline etk-analyze/.z3-trace:1]: What's this?

**Review by SamWilsn — APPROVED** [review]

File: etk-analyze/.z3-trace
Comment (lightclient) [reply etk-analyze/.z3-trace:1]: oof

Comment (lightclient) [reply etk-analyze/.z3-trace:1]: deleted

**Pull request merged**